### PR TITLE
#185: Fix @path in tapiro (closes #185)

### LINKED ISSUE #185: Fix @path in tapiro
`@path` annotation is currently broken in tapiro. However, this is pretty useful in general. 

Currently, metarpheus doesn't provide both `@path` and `controllerName`. Tapir requires both data.

### PR DESCRIPTION
**Pull request by calippo**
Closes #185

### tests performed

> A Test Plan is used to show what you tested to make sure your code works fine. You should write here all that you did to test, and provide some results of your testing.

> These results can be screenshots, query results, or even just pointers to unit tests that successfully passed.

### tests not performed (domain coverage)

> At times not everything can be tested, and writing what hasn't been tested is just as important as writing what has been tested.

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.

### CODE REVIEW

File: metarpheus/core/src/main/scala/io.buildo.metarpheus/core/extractors/controller.scala [hunk 1]
```diff
@@ -87,9 +90,11 @@ package object controller {
...
     }
 
-    val (controllerName, name) = t.mods.collectFirst {
-      case Mod.Annot(Init(Name("path"), _, Seq(Seq(Lit.String(n))))) => (t.name.value, n)
-    }.getOrElse((t.name.value, t.name.value))
+    val controllerName = t.name.value
+    val pathName = t.mods.collectFirst {
+      case Mod.Annot(Init(Name("path"), _, Seq(Seq(Lit.String(n))))) => n
+    }
+    val segmentPathName = pathName.getOrElse(t.name.value)
```
Comment (gabro) [inline metarpheus/core/src/main/scala/io.buildo.metarpheus/core/extractors/controller.scala]:
```suggestion
    val pathPrefix = pathName.getOrElse(t.name.value)
```

File: tapiro/core/src/main/scala/io/buildo/tapiro/AkkaHttpMeta.scala [hunk 2]
```diff
@@ -9,13 +9,12 @@ object AkkaHttpMeta {
...
     controllerName: Type.Name,
-    endpointsName: Term.Name,
+    httpEndpointsName: Term.Name,
     implicits: List[Term.Param],
     akkaHttpEndpoints: List[Defn.Val],
     routes: Term,
   ) => {
-    val tapirEndpoints = q"val endpoints = $endpointsName.create[AuthToken](statusCodes)"
-    val akkaHttpEndpointsName = Term.Name(s"${controllerName.syntax}AkkaHttpEndpoints")
+    val tapirEndpoints = q"val endpoints = $httpEndpointsName.create[AuthToken](statusCodes)"
```
Comment (tpetrucciani) [inline tapiro/core/src/main/scala/io/buildo/tapiro/AkkaHttpMeta.scala]: I was trying this out, I think here `httpEndpointsName` should be instead `tapirEndpointsName` (which must be added as as another parameter)

(and the same for http4s)

**Comment by tpetrucciani** [comment]
I've added two commits (one for my previous comment, one for the tests)

@calippo can you have a look? (I can't assign you to review since it's your PR)

The tests pass locally, but they fail on the CI apparently because `curl` is not available
```
[info] Checking if localhost:8080 is up...
[info] [error] java.io.IOException: Cannot run program "curl": error=2, No such file or directory
```

**Comment by calippo** [comment]
lgtm, ask @gabro for ci

**Comment by tpetrucciani** [comment]
👍 I'll merge this, I've created a new issue for the CI problem (#208)

**Pull request merged**